pkg/ccl/changefeedccl/changefeed_processors.go

```diff
@@ -1295,7 +1314,50 @@ func (cf *changeFrontier) Start(ctx context.Context) {
 	}()
 }
 
+func (cf *changeFrontier) runBillingMetricReporting(ctx context.Context) {
+	if cf.spec.JobID == 0 { // don't bill for core (non-enterprise) changefeeds
+		return
+	}
+
+	// Deregister the billing metric when the goroutine exits so we don't overbill.
+	defer cf.metrics.PerFeedAggMetrics.DeregisterJob(cf.spec.JobID)
+
+	var t timeutil.Timer
+	defer t.Stop()
+	for ctx.Err() == nil {
+		ivl := jitter(changefeedbase.BillingMetricsReportingInterval.Get(&cf.flowCtx.Cfg.Settings.SV))
+		start := timeutil.Now()
+
+		// Set a timeout of half the ivl so we still have a duty cycle in degenerate scenarios. This is just a guess though..
+		reqCtx, cancel := context.WithTimeout(ctx, ivl/2)
```

Comment:
This is a bit of a shot in the dark. Would like to hear opinions on how we like to do this.